Erin's solutions

eksmith opened the pull request:
@nicolasmccurdy 

Here are my solutions for LetsDrill. I would love feedback on the following and any other suggestions you think of: 

  - I am not a fan of the way I wrote the case; I don't think it's very efficient. 
  - I would appreciate any refactoring suggestions for shortest_string. 

Thanks!

degoeym commented:
Overall: your code works, so kudos on that. :tada: 

The case statement *definitely* isn't as clean/short as it could be. Do a bit of research on Ruby case switching to see what you can find to tidy it up a bit. One big hint: you don't need to write the variable name on each of your whens.

There are other enumerables that might be a more efficient/shorter, but refactoring what you have isn't that big of a deal, in my opinion anyway.

degoeym commented:
Also, because Jay will mention it, Ruby convention is two spaces for indentation of new lines.

eksmith commented:
@degoeym Those are all awesome points. Thank you so much for taking the time to write me feedback. I addressed the issues and hopefully everything looks a lot cleaner. 

I fixed indentation as well. Stupid Sublime keeps switching it back to 4 after I set it. I definitely have plans to dump it for Vim soon. 

Thanks again. I really appreciate it.

degoeym commented:
:tada: :tada: 

No problem. Not sure why Sublime would do that. Could also check out the new hotness Atom editor. I've been using it for a couple months now and like it. Feels way more extensible than Sublime.

nicolasmccurdy commented:
I have some experience with taming editor settings, let me know if you need help with setting up indentation on Sublime or another editor.

nicolasmccurdy commented on letsdrill.rb:
```diff
@@ -1,17 +1,23 @@
-
-  #Put your code here!
-
+	array.min_by {|string| string.length}
```
You can make this shorter with the `&:` symbol syntax:

``` ruby
array.min_by(&:length)
```

nicolasmccurdy commented:
:+1: Nice solution!

jaybobo commented:
:+1:
